paasta_tools/kubernetes_tools.py

```diff
@@ -1139,6 +1150,26 @@ def __init__(self) -> None:
         self.autoscaling = kube_client.AutoscalingV2beta1Api()
 
 
+def force_delete_pods(
+    service: str,
+    paasta_service: str,
+    instance: str,
+    namespace: str,
+    kube_client: KubeClient,
+):
+    all_pods = pods_for_service_instance(paasta_service, instance, kube_client)
+    pods_to_delete = filter_pods_by_service_instance(all_pods, paasta_service, instance)
```

Comment:
I don't think you need to filter. You have already filtered pods by service and instance in pods_for_service_instance.